File: 2023/advent_of_code/test/q21/q21_test.exs
defmodule AdventOfCode.Q21Test do
  use ExUnit.Case
  doctest AdventOfCode.Q21

  test "Q21 Part 1 Example works" do
    file_stream = File.stream!("./test/q21/q21_example.txt")
    result = AdventOfCode.Q21.part1(file_stream, 6)
    assert result == 16
  end

  test "Q21 Part 1 works" do
    file_stream = File.stream!("./test/q21/q21.txt")
    result = AdventOfCode.Q21.part1(file_stream)
    assert result == 3724
  end

  test "Q21 Part 2 works" do
    file_stream = File.stream!("./test/q21/q21.txt")
    result = AdventOfCode.Q21.part2(file_stream)
    assert result == 620_348_631_910_321
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q21/Q21Test.kt
package q21

import junit.framework.TestCase.assertEquals
import java.nio.file.Paths
import kotlin.test.Test

class Q21Test {
    @Test
    fun `part 1 works`()  {
        val filePath = javaClass.getResource("/q21.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(93813115694560, part1(filePath!!) )
    }

    @Test
    fun `part 2 works`()  {
        val filePath = javaClass.getResource("/q21.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(3910938071092, part2(filePath!!) )
    }
}